core/src/Kokkos_Parallel.hpp

```diff
@@ -496,6 +496,96 @@ void parallel_scan( const std::string& str
   (void) str;
 }
 
+//VINH DANG -- Adding the following for returning final scan result
+template< class ExecutionPolicy , class FunctorType, class ReturnType >
+inline
+void parallel_scan( const ExecutionPolicy & policy
+                  , const FunctorType     & functor
+                  , const std::string& str = ""
```

Comment:
The `str` argument should be the second to last argument, after `return_value`.

Reply:
This is not yet resolved ...